near-contract-standards/src/fungible_token/core_impl.rs

```diff
@@ -185,21 +189,35 @@ impl FungibleToken {
             let receiver_balance = self.accounts.get(&receiver_id).unwrap_or(0);
             if receiver_balance > 0 {
                 let refund_amount = std::cmp::min(receiver_balance, unused_amount);
-                self.accounts.insert(&receiver_id, &(receiver_balance - refund_amount));
+                if let Some(new_receiver_balance) = receiver_balance.checked_sub(refund_amount) {
+                    self.accounts.insert(&receiver_id, &new_receiver_balance);
+                } else {
+                    env::panic_str("The receiver account doesn't have enough balance");
+                }
 
                 if let Some(sender_balance) = self.accounts.get(sender_id) {
-                    self.accounts.insert(sender_id, &(sender_balance + refund_amount));
+                    if let Some(new_sender_balance) = sender_balance.checked_add(refund_amount) {
+                        self.accounts.insert(sender_id, &new_sender_balance);
+                    } else {
+                        env::panic_str("Sender balance overflow");
+                    }
+
                     FtTransfer {
                         old_owner_id: &receiver_id,
                         new_owner_id: sender_id,
                         amount: &U128(refund_amount),
                         memo: Some("refund"),
                     }
                     .emit();
-                    return (amount - refund_amount, 0);
+                    let used_amount = amount
+                        .checked_sub(refund_amount)
+                        .unwrap_or_else(|| env::panic_str("Total supply overflow"));
+                    return (used_amount, 0);
                 } else {
                     // Sender's account was deleted, so we need to burn tokens.
-                    self.total_supply -= refund_amount;
+                    self.total_supply
+                        .checked_sub(refund_amount)
+                        .unwrap_or_else(|| env::panic_str("Total supply overflow"));
```

Comment:
this isn't subtracting from `self.total_supply`, so this is a bug

Reply:
Good catch, fixed in 261efd01.
Please note this actually also means that this part is not enough covered with the unit tests.